packages/adblocker/src/engine/engine.ts

```diff
@@ -1443,29 +1443,70 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
     if (request.isSupported) {
       // Check the filters in the following order:
       // 1. $important (not subject to exceptions)
-      // 2. redirection ($redirect=resource)
+      // 2. redirection ($removeparam and $redirect=resource)
       // 3. normal filters
       // 4. exceptions
       result.filter = this.importants.match(request, this.isFilterExcluded.bind(this));
 
       let redirectNone: NetworkFilter | undefined;
       let redirectRule: NetworkFilter | undefined;
+      let redirectUrl: string | undefined;
 
       // If `result.filter` is `undefined`, it means there was no $important
-      // filter found so far. We look for a $redirect filter.  There is some
-      // extra logic to handle special cases like redirect-rule and
-      // redirect=none.
-      //
-      // * If redirect=none is found, then cancel all redirects.
-      // * Else if redirect-rule is found, only redirect if request would be blocked.
-      // * Else if redirect is found, redirect.
+      // filter found so far. We look for $removeparam and $redirect filter.
       if (result.filter === undefined) {
-        const redirects = this.redirects
-          .matchAll(request, this.isFilterExcluded.bind(this))
-          // highest priorty wins
-          .sort((a, b) => b.getRedirectPriority() - a.getRedirectPriority());
+        const redirectFilters = this.redirects.matchAll(request, this.isFilterExcluded.bind(this));
+        const resourceRedirects: NetworkFilter[] = [];
+        const requestRedirects: NetworkFilter[] = [];
+        for (const filter of redirectFilters) {
+          if (filter.isRemoveParam()) {
+            requestRedirects.push(filter);
+          } else {
+            resourceRedirects.push(filter);
+          }
+        }
+
+        // We don't need to match `removeparam` at all if:
+        // * `removeparam` filters not matched
+        // * URL doesn't have search parameter
+        if (requestRedirects.length !== 0) {
+          const url = new URL(request.url);
+          if (url.searchParams.size !== 0) {
+            for (const filter of requestRedirects) {
+              const key = filter.removeparam!;
+
+              // Remove all params in case of option value is empty
+              if (key === '') {
+                result.filter = filter;
+                redirectUrl = request.url.slice(0, request.url.indexOf('?'));
```

Comment:
This will also remove the anchors `#` at the end of the URL

Reply:
The part after `#` (hashtag fragment) is not sent to the server at all and shouldn't be included there.